manager_dashboard/manager_dashboard/js/usergroup-management.js

```diff
@@ -0,0 +1,150 @@
+// Clear input values when modal is to be shown
+$("#add-usergroup-modal").on('show.bs.modal', function() {
+  $('#usergroup-name').val('');
+  $('#usergroup-description').val('');
+});
+
+// Shows loading message in the modal and disables the inputs
+function showUserGroupLoading() {
+  $('#add-usergroup-modal-loading-message-container').removeClass('hidden');
+  $('#usergroup-name').attr('disabled', true);
+  $('#usergroup-description').attr('disabled', true);
+}
+
+// Hides loading message in the modal and enables the inputs
+function hideUserGroupLoading() {
+  $('#usergroup-name').attr('disabled', false);
+  $('#usergroup-description').attr('disabled', false);
+  $('#add-usergroup-modal-loading-message-container').addClass('hidden');
+}
+
+// Encodes the given string to the firebase key compatible string
+function encodeStringToKey(string) {
+  return encodeURIComponent(string).replace(/\./g, '%2E');
+}
+
+// Gets triggered when user hits submit button in add usergroup modal
+function submitNewUserGroup() {
+  const nameInput = $('#usergroup-name');
+  const name = (nameInput.val() || '').trim();
+  const descriptionInput = $('#usergroup-description');
+  const description = (descriptionInput.val() || '').trim();
+
+  if (name && name.length > 30) {
+    hideUserGroupLoading();
+    alert('User group name is required and should be less than 30 characters');
+  } else if (description && description.length > 100) {
+    hideUserGroupLoading();
+    alert('User group description is should be less than 100 characters');
+  } else {
+    showUserGroupLoading();
+
+    var key;
+    try {
+      key = encodeStringToKey(name);
+    } catch {
+      // This should never occur
+      alert('Failed to encode name into a key, please make sure that the user group name does not contain any invalid characters.');
```

Comment:
Your typical user probably won't know what characters are not valid for firebase keys :) Maybe you could add a list of what is allowed (or not, depending on what's shortest).